src-built-in/components/authentication/app.jsx

```diff
@@ -0,0 +1,93 @@
+/*!
+* The authentication component is a React component that renders a user form to collect auth info and determine whether to give user access to the rest of finsemble
+* based on the information provided. 
+* Copyright 2020 by ChartIQ, Inc.
+* All rights reserved.
+*/
+
+import React, { useState } from "react";
+import ReactDOM from "react-dom";
+import { AuthHeader, AuthForm, ErrorMessage, AuthInput, AuthSubmit } from "@chartiq/finsemble-ui/lib/components";
+import useAuth from "@chartiq/finsemble-ui/lib/hooks/useAuth";
+
+export const Authentication = () => {
+    const { handleChange, formValues, proceedToAuthorize, quitApplication } = useAuth();
+    const { error, setError } = useState(null);
+
+    /*
+     * Replace this function with your own authentication method that makes the call to your authentication server.
+     */
+    const authenticateUser = async () => {
+        // The user form data to be sent to the authentication server.
+        const { username, password } = formValues;
+        // Replace this with your own authentication calls. Currently it will always authenticate no matter what the user input is.
+        const result = await Promise.resolve({res: "ok"});
+        // the result is an object containing a response and an error. For example:
+        // {
+        //     res: "ok",
+        //     error: "Password is not correct"
+        // }
+        return result;
+    }
+
+    /**
+     * This function invokes the function that makes the call to the server that verifies user authentication information and
+     * decides whether to authorize the user to the rest of the application or display an error on the authentication form.
+     * @param {func} authenticateUser The function to call to authenticate the user.
+     */
+    const onSubmit = async (authenticateUser) => {
+        const authResult = await authenticateUser();
+        if (authResult.res === "ok") {
+            proceedToAuthorize();
+        } else {
+            setError(authResult.error);
+        }
+    };
+
+    return (
+        <>
+            <AuthHeader>
+                <img className="fsbl-logo" />
+                <div className="fsbl-close">
+                    <i className="ff-close" onClick={quitApplication}></i>
+                </div>
+            </AuthHeader>
+
+            <AuthForm>
+                <div className="fsbl-login-logo" alt="Finsemble Sign In"></div>
+                <div className="fsbl-button-wrapper">
+                    {error && <ErrorMessage>
+                        {error}
+                    </ErrorMessage>}
+                    <AuthInput
+                        type="text"
+                        name="username"
+                        placeholder="Username"
+                        onChange={handleChange} 
```

Comment:
What kind of customizations do we expect users to make to this function?

Do they need to cancel/modify input? If not, then we should call `handleChange` inside `AuthInput` and make `onChange` optional, instead of requiring they always pass the same function in.

Reply:
I'm sorry I didn't bring this up in the design - I didn't get a chance to look at the doc.

Reply:
We need to capture the form input change and store them inside formValues for sending the information to the auth server. This is to make it a React controlled form input element. I've changed the function name to make its purpose clearer.